docs/transactions/ensuring-data-integrity.md

```diff
@@ -0,0 +1,28 @@
+---
+sidebar_position: 50
+title: Ensuring Data Integrity
+---
+
+Transactions in Macrometa GDN are executed initially in main memory. The outcome of these transactions can be either a rollback or a commit. In case of a rollback, data is not written to disk; instead, the operations from the transaction are reversed in memory.
+
+Macrometa allows users to choose between full durability and delayed synchronization in single collection transactions, balancing throughput and performance against the risk of data loss.
+
+In a server crash scenario, multi-collection transactions that are uncommitted or in the process of being committed are rolled back upon server restart.
+
+## Committing Transactions
+
+When a transaction is committed, all modifications made within it are written to the collection data files. If the modified collections have the `waitForSync` property set to _true_ or if any operation in the transaction was executed with the `waitForSync` attribute, then these writes are synchronized to disk.
+
+## Durability in Multi-Collection Transactions
+
+Transactions involving multiple collections automatically synchronize to disk to ensure both durability and consistency in case of a server crash.
```

Comment:
Could you please provide the source of this information? I'm not aware of that.

Reply:
Old Transactions doc: https://drive.google.com/drive/folders/1RfsaIJhwo7MI7butDtTqQmc8bf7OKwHF

Reply:
Granted.

Reply:
No. I was told that this version of transactions is the same as the old version, and this is the old documentation. That's all I know.

Reply:
@diana-macrometa, I'm still waiting for Martin's or Mahesh's answer regarding all the information related to single-collection transactions stated in this page. Both of them are out this week though.

I don't know who the author of the `durability.md` document is, but the examples on it are JavaScript transactions, so I wonder if the single-collection transaction information is JavaScript-specific. If this is the case, then we will have to remove all of the single-collection related parts from this page.

Reply:
I'm also unaware of the correctness of the statement - `Transactions involving multiple collections automatically synchronize to disk to ensure both durability and consistency in case of a server crash.`

Reply:
So, sounds like I should just remove that sentence. Anything else?

Reply:
I think the `Single-Collection Transactions` section can also be removed as we don't have confirmation regarding authenticity of information under it. Rest of the PR looks good to me.

Reply:
Removed. @luisleite-macrometa , anything else?